iceoryx2-bb/lock-free/src/mpmc/unique_index_set.rs

```diff
@@ -554,12 +599,35 @@ impl<const CAPACITY: usize> FixedSizeUniqueIndexSet<CAPACITY> {
     ///    [`FixedSizeUniqueIndexSet::acquire_raw_index()`]
     ///  * The index should not be released twice
     ///
-    pub unsafe fn release_raw_index(&self, index: u32) {
-        self.state.release_raw_index(index)
+    pub unsafe fn release_raw_index(&self, index: u32, mode: ReleaseMode) -> ReleaseState {
+        self.state.release_raw_index(index, mode)
     }
 
     /// Returns the current len.
     pub fn borrowed_indices(&self) -> usize {
         self.state.borrowed_indices()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use iceoryx2_bb_testing::assert_that;
+
+    use super::HeadDetails;
+
+    #[test]
+    fn head_details() {
+        let sut_value = HeadDetails {
+            head: 12345,
+            aba: 6789,
+            borrowed_indices: 54321,
+        }
+        .value();
+
+        let sut = HeadDetails::from(sut_value);
+
+        assert_that!(sut.head, eq 12345);
+        assert_that!(sut.aba, eq 6789);
+        assert_that!(sut.borrowed_indices, eq 54321);
+    }
```

Comment:
Ideally, there would be a check that the fields are not overlapping, e.g. subsequently setting each field to it's max value and checking if the others remain 0.